### DISCUSSION TITLE
Auto start VM upon app launch

### DISCUSSION THREAD

**Discussion by ScottEKendall** [utmapp/UTM#5913] [General]
Is it possible to have a VM auto launch when UTM launches?  I run Home Assistant in a VM and it would be really great to have the VM auto launch if I ever have to restart my Mac..

**Comment by reknef** [comment]
I have exactly the same question. 
Anybody??

**Comment by longzheng** [comment]
I'm not sure if there's a way to auto start a VM when UTM launches but it is possible to auto start a VM when the Mac logs in.

Start “**Automator**”. Create an “**Application**” workflow

![image](https://github.com/user-attachments/assets/2a8d952f-79b8-4863-971b-548a22983b1f)

Search for “Launch Application” action. Double click or drag it into the workflow to add. Select “Other” from the dropdown and then find “UTM.app”.

![image](https://github.com/user-attachments/assets/acf8ad39-9700-48aa-bac8-bcf0565bf106)

Search for “**Run Shell Script**” action. **Double click** or drag it into the workflow to add.

![image](https://github.com/user-attachments/assets/54d052e2-5ccb-4b2e-99d3-eed82574b631)

**Change the shell script** to (replace "Home Assistant" with the name of your Home Assistant VM)
```
/Applications/UTM.app/Contents/MacOS/utmctl start "Home Assistant"
```

![image](https://github.com/user-attachments/assets/eacad260-7041-4d3f-b28f-33b3bbf62ca5)

To test the workflow, make sure UTM is not running or stop the VM. Click **“Run” command in the top right of Automator** and **approve** the permission prompt

![image](https://github.com/user-attachments/assets/604ae01c-0678-4d51-9be8-2870304fe1e6)

Verify UTM starts the VM.

![image](https://github.com/user-attachments/assets/dd149eac-e411-4896-8b2c-3876e5ebded0)

**Save** the Automator workflow as “UTM Start Home Assistant.app”

![image](https://github.com/user-attachments/assets/8de3d961-4f17-46b2-a512-486039ed5058)

Open “**Login Items**” in “Users & Groups” settings

**Drag or add** the “UTM Start Home Assistant” application into the Login Items (the list UI might be buggy and doesn’t show it immediately, switch between the “Password” and “Login Items” tabs to refresh the list)

![image](https://github.com/user-attachments/assets/ed313962-80cd-44f6-8917-07b3473f9766)

**Restart** the computer. The first time the workflow runs there will be another permissions prompt, **approve**. Afterwards it shouldn’t prompt anymore.

**Comment by RafBois** [comment]
Has anyone tried what @longzheng is suggesting?

It seems that not many people are asking for this feature, which I find essential.

I'm using a Mac mini M1 as a home automation server (due to its performance and low power consumption). I would like certain VMs to restart automatically after a power outage or a macOS update, without having to bother creating specific scripts for each machine I'm testing.

**Reply by henryxue25** [reply]
Just added automator script to MAC mini M4 MAC OS26, works well (it only starts after login). Not sure if there is a way you can start the VM after MAC power on like a Windows Service.
